Data Structures/Linked-Lists/Implementations of single linked list.cpp

```diff
@@ -0,0 +1,379 @@
+//NAME = Rishabh Garg
+//EMAIL ID = rishabhgarg25699@gmail.com
+
+/*  Program of single linked list and various implementations  */
+
+
+#include<stdio.h>
+#include<stdlib.h>
+
+struct node
+{
+	int info;
+	struct node *link;
+};
+
+struct node *create_List(struct node *start);
+void display(struct node *start);
+void count(struct node *start);
+void search(struct node *start,int data);
+struct node *addAtBeg(struct node *start,int data);
+struct node *addAtEnd(struct node *start,int data);
+struct node *addAfter(struct node *start,int data,int item);
+struct node *addBefore(struct node *start,int data,int item );
+struct node *addAtPos(struct node *start,int data,int pos);
+struct node *del(struct node *start,int data);
+struct node *reverse(struct node *start);
+ 
+//MAIN FUNCTIOM
+
+int main()
+{
+	struct node *start=NULL;	
+	int choice,data,item,pos;
+		
+	while(1)
+	{
+		printf("1.Create List\n");
+		printf("2.Display\n");
+		printf("3.Count\n");
+		printf("4.Search\n");
+		printf("5.Add to empty list / Add at beginning\n");
+		printf("6.Add at end\n");
+		printf("7.Add after node\n");
+		printf("8.Add before node\n");
+		printf("9.Add at position\n");
+		printf("10.Delete\n");
+		printf("11.Reverse\n");
+		printf("12.Quit\n\n");
+		
+		printf("Enter your choice : ");
+		scanf("%d",&choice);
+		
+		switch(choice)
+		{
+		 case 1:
+			start=create_List(start);
+			break;
+		 case 2:
+			display(start);
+			break;
+		 case 3:
+			count(start);
+			break;
+		 case 4:
+			printf("Enter the element to be searched : ");
+			scanf("%d",&data);
+			search(start,data);
+			break;
+		 case 5:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			start=addAtBeg(start,data);
+			break;
+		 case 6:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			start=addAtEnd(start,data);
+			break;
+		 case 7:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the element after which to insert : ");
+			scanf("%d",&item);
+			start=addAfter(start,data,item);
+			break;
+		 case 8:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the element before which to insert : ");
+			scanf("%d",&item);
+			start=addBefore(start,data,item);
+			break;
+		 case 9:
+			printf("Enter the element to be inserted : ");
+			scanf("%d",&data);
+			printf("Enter the position at which to insert : ");
+			scanf("%d",&pos);
+			start=addAtPos(start,data,pos);
+			break;
+		 case 10:
+			printf("Enter the element to be deleted : ");
+			scanf("%d",&data);
+			start=del(start, data);	
+			break;
+		 case 11:
+			start=reverse(start);
+			break;
+		 case 12:
+			 exit(1);
+		 default:
+			 printf("Wrong choice\n");
+		}/*End of switch */
+	}/*End of while */
+
+return 0;
+
+}
+
+/*End of main()*/
+
+
+struct node *create_List(struct node *start)

```

Comment:
Add a brief explanation of what this function does. Fix it in the complete code.